### ISSUE TITLE
Training with model: to_numpy: true

### ISSUE THREAD

**Issue by infophysics** [DeepLearnPhysics/spine#26]
> ## I'm submitting a ...
> 
> * [x]  bug report
> * [ ]  feature request
> 
> ## Expected Behavior
> Careful when trying to run a training with to_numpy being set to true as this will remove the gradients required for backprop:
![image](https://github.com/user-attachments/assets/d5bc98b7-1e61-4d2e-ae93-679acf156f2e)

> 
> ## Current Behavior
> You may see an error like this:
![image](https://github.com/user-attachments/assets/de1895d1-dcf6-4c9e-b3fa-7eda7ea7f576)

> 
> ## How to reproduce
> Try setting to_numpy to true in any training script.
> 
> ## Possible Solution (optional)
> Obviously comment out the to_numpy or set to false.
> 
> ## Checklist
> Must provide the following (check when done):
> 
> * [x]  I've added at least 1 label
> * [ ]  Assigned a reviewer

**Comment by francois-drielsma** [comment]
Configuration-based solution provided above. When casting using `to_numpy: true`, the output of a model loses its gradients and as such the model cannot be trained.

The issue has been resolved by #27 the `loss` value to python float **after** the backward function is called.
